src/components/container/container.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Suspense, useCallback, useEffect } from 'react';
+import { Outlet, useNavigate } from 'react-router-dom';
+
+import { useWallet } from '@common/hooks/use-wallet';
+import { RouteUrls } from '@routes/route-urls';
+import { useAuthRequest } from '@store/onboarding/onboarding.hooks';
+import { usePendingTransaction } from '@store/transactions/transaction.hooks';
+import { useOnCancel } from '@store/transactions/requests.hooks';
+import { useRouteHeaderState } from '@store/ui/ui.hooks';
+
+import { ContainerLayout } from './container.layout';
+
+function UnmountEffectSuspense() {
+  const pendingTx = usePendingTransaction();
+  const { authRequest } = useAuthRequest();
+  const handleCancelTransaction = useOnCancel();
+  const { cancelAuthentication } = useWallet();
+
+  /*
+   * When the popup is closed, this checks the request type and forces
+   * the request promise to fail; triggering an onCancel callback function.
+   */
+  const handleUnmount = useCallback(async () => {
+    if (!!pendingTx) {
+      await handleCancelTransaction();
+    } else if (!!authRequest) {
+      cancelAuthentication();
+    }
+  }, [cancelAuthentication, authRequest, pendingTx, handleCancelTransaction]);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', handleUnmount);
+    return () => window.removeEventListener('beforeunload', handleUnmount);
+  }, [handleUnmount]);
+
+  return null;
+}
+
+function UnmountEffect() {
+  return (
+    <Suspense fallback={<></>}>
+      <UnmountEffectSuspense />
+    </Suspense>
+  );
+}
+
+export function Container(): JSX.Element | null {
+  const { hasGeneratedWallet, hasRehydratedVault } = useWallet();
+  const [routeHeader, _] = useRouteHeaderState();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // This ensures the ext popup hits the right route on load
+    if (!hasGeneratedWallet) navigate(RouteUrls.Onboarding);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return hasRehydratedVault ? (
+    <ContainerLayout header={routeHeader}>
+      <UnmountEffect />
+      <Outlet />
+    </ContainerLayout>
+  ) : null;
```

Comment:
Could this ternary be in the routes file instead?

Reply:
Yep, I'm sure it can be.

Reply:
This did NOT end up behaving the same so I left it as-is.

Reply:
Nvm! I did refactor this and it is working just fine!